src/components/application_manager/test/commands/mobile/on_system_request_notification_test.cc

```diff
@@ -0,0 +1,209 @@
+using namespace mobile_apis;
+using testing::Return;
+using testing::ReturnRef;
+using testing::_;
+
+namespace {
+const uint32_t kConnectionKey = 1u;
+}  // namespace
+
+class OnSystemRequestNotificationTest
+    : public CommandsTest<CommandsTestMocks::kIsNice> {
+ public:
+  void SetUp() OVERRIDE {
+    message_helper_ =
+        application_manager::MockMessageHelper::message_helper_mock();
+  }
+
+  MockMessageHelper* message_helper_;
+};
+
+TEST_F(OnSystemRequestNotificationTest, Run_ProprietaryType_SUCCESS) {
+  const RequestType::eType kRequestType = RequestType::PROPRIETARY;
+
+  MessageSharedPtr msg = CreateMessage();
+  (*msg)[strings::params][strings::connection_key] = kConnectionKey;
+  (*msg)[strings::msg_params][strings::request_type] = kRequestType;
+
+  SharedPtr<OnSystemRequestNotification> command =
+      CreateCommand<OnSystemRequestNotification>(msg);
+
+  MockAppPtr mock_app = CreateMockApp();
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app));
+  MockPolicyHandlerInterface mock_policy_handler;
+  EXPECT_CALL(app_mngr_, GetPolicyHandler())
+      .WillRepeatedly(ReturnRef(mock_policy_handler));
+  std::string string;
+  EXPECT_CALL(*mock_app, policy_app_id()).WillOnce(Return(string));
+  EXPECT_CALL(mock_policy_handler, IsRequestTypeAllowed(_, _))
+      .WillOnce(Return(true));
+
+#ifdef EXTENDED_POLICY
+  EXPECT_CALL(app_mngr_, GetPolicyHandler())
+      .Times(2)
+      .WillRepeatedly(ReturnRef(mock_policy_handler));
+  EXPECT_CALL(mock_policy_handler, TimeoutExchange()).WillOnce(Return(5u));
+#endif
+
+  EXPECT_CALL(*message_helper_, PrintSmartObject(_)).WillOnce(Return(false));
+  EXPECT_CALL(app_mngr_, SendMessageToMobile(msg, _));
+
+  command->Run();
+
+  ASSERT_EQ(FileType::JSON,
+            (*msg)[strings::msg_params][strings::file_type].asInt());
+  ASSERT_EQ(application_manager::MessageType::kNotification,
+            (*msg)[strings::params][strings::message_type].asInt());
+  ASSERT_EQ(CommandImpl::mobile_protocol_type_,
+            (*msg)[strings::params][strings::protocol_type].asInt());
+  ASSERT_EQ(CommandImpl::protocol_version_,
+            (*msg)[strings::params][strings::protocol_version].asInt());
+}
+
+TEST_F(OnSystemRequestNotificationTest, Run_HTTPType_SUCCESS) {
+  const RequestType::eType kRequestType = RequestType::HTTP;
+
+  MessageSharedPtr msg = CreateMessage();
+  (*msg)[strings::params][strings::connection_key] = kConnectionKey;
+  (*msg)[strings::msg_params][strings::request_type] = kRequestType;
+
+  SharedPtr<OnSystemRequestNotification> command =
+      CreateCommand<OnSystemRequestNotification>(msg);
+
+  MockAppPtr mock_app = CreateMockApp();
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app));
+  MockPolicyHandlerInterface mock_policy_handler;
+  EXPECT_CALL(app_mngr_, GetPolicyHandler())
+      .WillOnce(ReturnRef(mock_policy_handler));
+  std::string string;
```

Comment:
@pvvasilev , can you name it more specific... policy_app_id or empty_policy_app_id ?
And check other files